spec/requests/graphql/mutations/term/create_term_spec.rb

```diff
@@ -0,0 +1,253 @@
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            prefLabel: 'Hippogriff',
+            altLabels: ['Hippogryph'],
+            termType: 'local',
+            customFields: [
+              { field: 'classification', value: 'Eagle' }
+            ]
+          }
+        }
+      end
+      let(:expected_response) do
+        %({
+            "term": {
+              "prefLabel": "Hippogriff",
+              "altLabels": ["Hippogryph"],
+              "termType": "local",
+              "authority": null,
+              "customFields": [
+                { "field": "classification", "value": "Eagle" },
+                { "field": "harry_potter_reference", "value": null }
+              ]
+            }
+          })
+      end
+
+      before { graphql query, variables }
+
+      it 'creates a term record' do
+        expect(Term.count).to be 1
+        expect(Term.first.uri).not_to be_blank
+        expect(Term.first.alt_labels.first).to eql 'Hippogryph'
+      end
+
+      it 'returns newly created term in json' do
+        expect(response.body).to be_json_eql(expected_response).at_path('data/createTerm').excluding('uri')
+      end
+    end
+
+    context 'when successfully creating a new temporary term' do
+      let(:variables) do
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            prefLabel: 'Hippogriff',
+            termType: 'temporary'
+          }
+        }
+      end
+
+      let(:expected_json) do
+        %({
+            "term": {
+              "prefLabel": "Hippogriff",
+              "altLabels": [],
+              "authority": null,
+              "termType": "temporary",
+              "customFields": [
+                { "field": "classification", "value": null },
+                { "field": "harry_potter_reference", "value": null }
+              ]
+            }
+          })
+      end
+
+      before { graphql query, variables }
+
+      it 'creates term record' do
+        expect(Term.count).to be 1
+        expect(Term.first.uri).to start_with('temp:')
+        expect(Term.first.pref_label).to eql 'Hippogriff'
+      end
+
+      it 'return newly created term in json' do
+        expect(response.body).to be_json_eql(expected_json).at_path('data/createTerm').excluding('uri')
+      end
+    end
+
+    context 'when uri is missing for external term' do
+      let(:variables) do
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            prefLabel: 'Minotaur (Greek mythological character)',
+            authority: 'fast',
+            termType: 'external',
+            customFields: [
+              { field: 'harry_potter_reference', value: false }
+            ]
+          }
+        }
+      end
+
+      before { graphql query, variables }
+
+      it 'returns error in json' do
+        expect(response.body).to be_json_eql(%(
+          "Uri can't be blank; Uri hash can't be blank"
```

Comment:
"Uri hash can't be blank" might be confusing for users, since they can't edit that value and aren't even aware that it exists.  Maybe we can update the model to remove the redundant validation error.  If you agree, I can ticket.

Reply:
I understand that its confusing but I don't think its a good idea to remove the validation from the model. That's the last check the model does before saving the record and if that check doesn't catch it then we will just be returning a database error.

Reply:
I didn't mean that we should remove the validation, just that it might be worth removing the redundant message when those two messages appear together, since if uri is blank, uri hash definitely won't be valid.  It's almost like saying "stop at the first error because that needs to be resolved before the second error can be properly evaluated".

Reply:
And even if we do skip the validation in the specific case when uri is blank, the validation is still essential as a protection if the code ever changes, paired with the db unique index you mentioned.

Reply:
ahh I see. We usually want to return multiple messages to the front end, for example, if you missed adding your first and last name that returns two errors. We generate these messages (validation errors) the same way for all models. I guess I am not sure how you envision implementing the change.

Reply:
In term.rb, can we add `if: Proc.new { |t| t.uri.present? }` to the end of the `validates :uri_hash` validation?

Reply:
Okay, feel free to ticket that work.

Reply:
Ticket created.